### ISSUE TITLE
Zensus cells without HH demand profiles

### ISSUE THREAD

**Issue by nesnoj** [openego/eGon-data#431]
Maybe I was too fast merging #260 :

Some Zensus cells with buildings and population do not have any profiles assigned. Let's consider this example plot (village: Husby, east of Flensburg):

![image](https://user-images.githubusercontent.com/15910173/132468257-0abb9aba-175b-4b1a-915b-f35f424a370b.png)

Legend:
- green cells: zensus cells **with** profiles assigned
- red cells: zensus cells **without** profiles assigned
- numbers: population and number of households from zensus
- brown: buildings

Obviously, there're no profiles assigned in cells without data on households - the methods do not cover those edge cases. In ding0 the buildings won't be part of the grid as they have no load. This might be acceptable in the example above but I've seen settlements with a handful of cells and only 1 holding profile information...

Is there a quick way of covering those edge cases, e.g. use the population (we had a similar case for a missing attribute, cf. [post](https://github.com/openego/eGon-data/pull/260#issuecomment-887586406)) and some average HH distribution?

**Comment by nailend** [comment]
I can think of the following:
1. identify cells with population but no households
2. count the buildings in these cells (maybe tag specifc? @nesnoj )
3. randomly choose profile-types for each building using the eurostat-distribution from the paper (von Appen, Haack,  Braun)

![Haushaltstypen](https://user-images.githubusercontent.com/51374526/132728006-70e69571-ed4a-40a6-83c3-25097d44c4d6.png)

You proposed to choose the amount of profiles by the population. As we will have to assign profiles to each building in #435 I would rather go by building-count. We will have to accept some inaccuracy but prevent further ad-hoc patches. 
What do you think?

**Comment by nesnoj** [comment]
Thanks for this proposal. 
I do see problems in using buildings:
- zensus: not all cells have building data 
- OSM: same here

So we'd have other edge cases here..

I'd say lets solve #435 first

**Comment by nailend** [comment]
Apparently we are missing about 20% of all cells with population.


|            | # cells | %    |
|------------|---------|------|
| population | 143521  | 100% |
| households | 113798  | 79,3 |
| difference | 29723   | 20,7 |

**Comment by nesnoj** [comment]
Damn it, 20% seems to be the magic number today @ClaraBuettner :chart_with_downwards_trend: .
Shall we use a average split hh/pop, e.g. from neighbouring cells / same municipality / ...?

**Comment by nesnoj** [comment]
Data for DE:

|            | # cells | %    |   people  |%    |
|------------|---------|------|------|------|
| population | 3177723  | 100 | 80324282| 100 | 
| population w/ households | 2659745  | 83,7 | 78017556 | 97,1 |
| population w/o households | 517978   | 16,3 | 2306726 | 2,9 |

Somewhat better..

**Comment by nesnoj** [comment]
I mean...[well](https://imgs.xkcd.com/comics/confidence_interval.png)...

**Comment by nesnoj** [comment]
We decided to fix this by allocating an appropriate mean household distribution to affected cells.
Start here

https://github.com/openego/eGon-data/blob/1a6206eaff4341c56dd870a2020016071c99b8cb/src/egon/data/datasets/electricity_demand_timeseries/hh_profiles.py#L1195-L1196

https://github.com/openego/eGon-data/blob/1a6206eaff4341c56dd870a2020016071c99b8cb/src/egon/data/datasets/electricity_demand_timeseries/hh_profiles.py#L878

**Comment by nailend** [comment]
> We decided to fix this by allocating an appropriate mean household distribution to affected cells as it affects both power and heat. 

After thinking about this, I realized that this is not trivial and a mean distribution does not bring a plausible solution for all cases. The population value of the cells would not fit to the amount of people which are allocated to these household types. Even if we determine a mean distribution just within subgroups of specific population values implausible deviations remain. Especially in the cells with very low population this would end up in household shares <1 and end up in a bunch of new edge cases to solve.

Taking the most common distribution within subgroups of specific population values seems to be the next closest but also generates more edge cases as soon as the subgroups become small and determining the most common is not trivial anymore.

Thus we decided to go for a random choice distribution within subgroups of specific population.
All cells which have a specific population value and no household distribution data, will get the same random household distribution applied which is picked out of cell with the same population value.
